### PR TITLE
Milestone-2

### PR DESCRIPTION
**Pull request by tacettinkustu**
In this step I configured my Redux Store and write actions and reducer for adding and removing books. I also added some UI elements to my components and make them dispatch actions. 

- Editing an existing book and updating the progress is not part of this step. Neither is the styling of the application's components.

### CODE REVIEW

File: src/components/books/AddBook.js [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
+  const handleSubmit = (event) => {
+    const book = {
+      id: nanoid(),
+      title,
+      category,
+    };
+    dispatch(addBook(book));
+    setTitle('');
+    setCategory(categories[0]);
+    event.preventDefault();
```
[the remark this answers is not in the document]
Comment (tacettinkustu) [reply src/components/books/AddBook.js:23]: It is working on me

[the remark this answers is not in the document]
Comment (tacettinkustu) [reply src/components/books/AddBook.js:23]: I checked

[the remark this answers is not in the document]
Comment (tacettinkustu) [reply src/components/books/AddBook.js:23]: It is working on my computer

File: src/components/books/Page.js [hunk 2]
```diff
@@ -1,41 +1,16 @@
...
-  {
-    category: 'Economy',
-    title: 'Capital in the Twenty-First Century',
-    author: 'Suzanne Collins',
-    progress: {
-      currentChapter: 'Introduction',
-      completed: '0',
-    },
-  },
-];
```
Comment (oxenprogrammer) [inline src/components/books/Page.js:32]: **Optional Changes**
The former Code Reviewer might have missed this.
- [ ] The initial State should be an empty `array`. By introducing objects in the array has added a layer of complexity to your code at the moment. I guess it is the reason you're finding it a little tedious to implement completed progress in the `BookProgress.js`

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi **@tacettinkustu** , 

Great work so far. You have put much energy in this and I encourage you to keep going 👏 
You can do this and I trust you.:muscle: 
There are still some changes that  aren't addressed yet. 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/books/Book.js [hunk 3]
```diff
@@ -24,18 +17,18 @@ const Book = ({ book }) => {
...
 
 Book.propTypes = {
   book: PropTypes.shape({
+    id: PropTypes.string.isRequired,
     title: PropTypes.string.isRequired,
     author: PropTypes.string.isRequired,
     category: PropTypes.string.isRequired,
     progress: Progress.propTypes.progress,
   }).isRequired,
 };
```
[the remark this answers is not in the document]
Comment (tacettinkustu) [reply src/components/books/Book.js:33]: I removed this sections for now.

**Review by AshakaE — APPROVED** [review]
### Hi @tacettinkustu  🙋‍♀️, 

# STATUS APPROVED ✔

Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**